### PR TITLE
refs(Incidents): Remove date_started automatic detection code and cleanup around removing manual incident creation

### PR DESCRIPTION
**Pull request by wedamija**
We no longer support manual incident creation, just removing some unused code around it.

### CODE REVIEW

**Comment by wedamija** [comment]
Migration is a no-op:
```
BEGIN;
--
-- Alter field type on incident
--
COMMIT;
```

File: src/sentry/incidents/models.py [hunk 1]
```diff
@@ -179,7 +178,6 @@ def snuba_values(self):
 
 
 class IncidentActivityType(Enum):
-    CREATED = 0
```
Comment (markstory) [inline src/sentry/incidents/models.py]: Are there any existing records in the database using these values that will be broken now?

Comment (wedamija) [reply src/sentry/incidents/models.py]: No, we flushed out the table and I've double checked - no existing records have this

**Review by markstory — APPROVED** [review]
Looks good tome.

**Pull request merged**